pkg/gather/clusterconfig/operators.go

```diff
@@ -92,28 +96,120 @@ func GatherClusterOperators(g *Gatherer, c chan<- gatherResult) {
 		c <- gatherResult{nil, []error{err}}
 		return
 	}
-	records, errors := gatherClusterOperators(g.ctx, gatherConfigClient, gatherKubeClient.CoreV1(), discoveryClient, dynamicClient)
-	c <- gatherResult{records, errors}
+	records, errs := gatherClusterOperators(g.ctx, gatherConfigClient, gatherKubeClient.CoreV1(), discoveryClient, dynamicClient)
+	c <- gatherResult{records, []error{errs}}
 }
 
-func gatherClusterOperators(ctx context.Context, configClient configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+// gatherClusterOperators collects cluster operators
+func gatherClusterOperators(ctx context.Context, configClient configv1client.ConfigV1Interface, coreClient corev1client.CoreV1Interface, discoveryClient discovery.DiscoveryInterface, dynamicClient dynamic.Interface) ([]record.Record, error) {
 	config, err := configClient.ClusterOperators().List(ctx, metav1.ListOptions{})
 	if errors.IsNotFound(err) {
 		return nil, nil
 	}
 	if err != nil {
-		return nil, []error{err}
+		return nil, err
 	}
+
+	// collect the cluster operators reports
+	records := clusterOperatorsRecords(ctx, config.Items, dynamicClient, discoveryClient)
+
+	// gather pods from unhealthy operators
+	pods, recs, totalContainers := gatherUnhealthyClusterOperator(ctx, config.Items, coreClient)
+	records = append(records, recs...)
+	// Exit early if no pods found
+	if len(pods) == 0 {
+		return records, nil
+	}
+
+	// gather pods containers logs
+	clogs, err := gatherPodContainersLogs(ctx, coreClient, pods, totalContainers)
+	if err != nil {
+		return records, nil
+	}
```

Comment:
This doesn't seem right.

Reply:
it should return the `err` as well IMO

Reply:
Yeah, I think that it should return the `err` and not return the `records` as well... Right? But, in this case, we first need to finish the discussion down below.

Reply:
BTW, now checking the previous/other returns on this exact same method. We always return the `records`... My thoughts would be... and if we continue returning the `records` but added a `log` event that logs the error when it occurred?